File: app/src/main/java/com/example/mvvmtest/manager/ZodiacManager.java
package com.example.mvvmtest.manager;

import com.example.mvvmtest.util.Zodiac;

public class ZodiacManager {

    public static Zodiac getZodiac(int day, int month) {
        switch (month) {
            case 1:
                if (day >= 21) {
                    return Zodiac.ACUARIO;
                } else {
                    return Zodiac.CAPRICORIO;
                }
            case 2:
                if (day >= 20) {
                    return Zodiac.PISCIS;
                } else {
                    return Zodiac.ACUARIO;
                }
            case 3:
                if (day >= 21) {
                    return Zodiac.ARIES;
                } else {
                    return Zodiac.PISCIS;
                }
            case 4:
                if (day >= 20) {
                    return Zodiac.TAURO;
                } else {
                    return Zodiac.ARIES;
                }
            case 5:
                if (day >= 21) {
                    return Zodiac.GEMINIS;
                } else {
                    return Zodiac.TAURO;
                }
            case 6:
                if (day >= 22) {
                    return Zodiac.CANCER;
                } else {
                    return Zodiac.GEMINIS;
                }
            case 7:
                if (day >= 23) {
                    return Zodiac.LEO;
                } else {
                    return Zodiac.CANCER;
                }
            case 8:
                if (day >= 24) {
                    return Zodiac.VIRGO;
                } else {
                    return Zodiac.LEO;
                }
            case 9:
                if (day >= 24) {
                    return Zodiac.LIBRA;
                } else {
                    return Zodiac.VIRGO;
                }
            case 10:
                if (day >= 23) {
                    return Zodiac.ESCORPIO;
                } else {
                    return Zodiac.LIBRA;
                }
            case 11:
                if (day >= 22) {
                    return Zodiac.SAGITARIO;
                } else {
                    return Zodiac.ESCORPIO;
                }
            case 12:
                if (day >= 22) {
                    return Zodiac.CAPRICORIO;
                } else {
                    return Zodiac.SAGITARIO;
                }
        }
        return null;
    }

}

File: app/src/main/java/com/example/mvvmtest/manager/ZodiacManager.kt
package com.example.mvvmtest.manager

import com.example.mvvmtest.util.Zodiac

object ZodiacManager {

    fun getZodiac(day: Int, month: Int): Zodiac? {
        when (month) {
            1 -> return if (day >= 21) {
                Zodiac.ACUARIO
            } else {
                Zodiac.CAPRICORIO
            }
            2 -> return if (day >= 20) {
                Zodiac.PISCIS
            } else {
                Zodiac.ACUARIO
            }
            3 -> return if (day >= 21) {
                Zodiac.ARIES
            } else {
                Zodiac.PISCIS
            }
            4 -> return if (day >= 20) {
                Zodiac.TAURO
            } else {
                Zodiac.ARIES
            }
            5 -> return if (day >= 21) {
                Zodiac.GEMINIS
            } else {
                Zodiac.TAURO
            }
            6 -> return if (day >= 22) {
                Zodiac.CANCER
            } else {
                Zodiac.GEMINIS
            }
            7 -> return if (day >= 23) {
                Zodiac.LEO
            } else {
                Zodiac.CANCER
            }
            8 -> return if (day >= 24) {
                Zodiac.VIRGO
            } else {
                Zodiac.LEO
            }
            9 -> return if (day >= 24) {
                Zodiac.LIBRA
            } else {
                Zodiac.VIRGO
            }
            10 -> return if (day >= 23) {
                Zodiac.ESCORPIO
            } else {
                Zodiac.LIBRA
            }
            11 -> return if (day >= 22) {
                Zodiac.SAGITARIO
            } else {
                Zodiac.ESCORPIO
            }
            12 -> return if (day >= 22) {
                Zodiac.CAPRICORIO
            } else {
                Zodiac.SAGITARIO
            }
        }
        return null
    }

}
